src/gen/lib/AST_grammar.ml

```diff
@@ -15,87 +15,22 @@ type rule_body =
   | Repeat of rule_body
   | Repeat1 of rule_body
   | Choice of rule_body list
+  | Optional of rule_body
```

Comment:
yep, good to infer it back. Actually in the original code we had this ast_grammar_normalized.ml (or .re) where the goal
was to normalize grammar.json in something better. This type had this OPTION too inferred back from CHOICE(....BLANK).

Reply:
I also lift CHOICE up, to generate regular variants.
type ident = string
  [@@deriving show]

type atom =
  | SYMBOL of ident 
  | TOKEN 
  | STRING of string 
  [@@deriving show { with_path = false }]

type simple =
  | ATOM of atom 
  | SEQ of simple list 
  | REPEAT of simple | REPEAT1 of simple
  | OPTION of simple 
  [@@deriving show { with_path = false }]

type rule_body =
  | CHOICE of simple list 
  | SIMPLE of simple 
  [@@deriving show { with_path = false }]

type rule = (ident * rule_body)

Reply:
Also here: https://github.com/returntocorp/ocaml-tree-sitter/blob/to_menhir/lib/ast_grammar_normalized2.ml

Reply:
I see. Not sure which is better in practice. I like a flat view until it gets unmanageable, like for example if we had 10 types of atoms rather than just 3.